test/src/main/java/org/axonframework/test/matchers/ExactClassMatcher.java

```diff
@@ -0,0 +1,38 @@
+package org.axonframework.test.matchers;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+
+import static org.axonframework.common.BuilderUtils.assertNonNull;
+
+/**
+ * Matcher testing for exact class
+ *
+ * @author Yoann CAPLAIN
+ * @since 4.6.2
+ */
+public class ExactClassMatcher<T> extends TypeSafeMatcher<T> {
+
+    private final Class<T> expectedClass;
+
+    public static <T> Matcher<T> exactClassOf(Class<T> expectedClass) {
+        return new ExactClassMatcher<T>(expectedClass);
+    }
+
+    public ExactClassMatcher(Class<T> expectedClass) {
```

Comment:
This `public` constructor missed some JavaDoc.

Reply:
JavaDoc added